### DISCUSSION TITLE
No coverage report data on windows-latest workflows

### DISCUSSION THREAD

**Discussion by stephane-caron** [ymyzk/tox-gh-actions#173] [Q&A] [answered]
**Describe the bug**
Since the release of tox-gh-actions 3.0.0 the qpsolvers project has been experiencing Windows CI issues https://github.com/stephane-caron/qpsolvers/issues/141. The symptom is:

```
py37-windows: commands[7]> coverage report --include=\"qpsolvers/*\"
  No data to report.
```

Which makes the CI fail. Version-pinning tox-gh-actions to 2.12.0 removes the symptom, as done e.g. in [this job](https://github.com/stephane-caron/qpsolvers/actions/runs/3781004586).

**To Reproduce**
* GitHub Actions configuration: [test.yml](https://github.com/stephane-caron/qpsolvers/blob/caa4bf50c4f6ccfd191b6e1459aaa8ec696fd727/.github/workflows/test.yml)
* tox and tox-gh-actions configuration: [tox.ini](https://github.com/stephane-caron/qpsolvers/blob/caa4bf50c4f6ccfd191b6e1459aaa8ec696fd727/tox.ini)
* Link to the GitHub Actions job if a repository is public: for instance [this one](https://github.com/stephane-caron/qpsolvers/actions/runs/3780566497/jobs/6426761600)

**Expected behavior**

Successful coverage report generation.

**Additional context**

* The issue only happens only on ``windows-latest`` images (``ubuntu-latest`` and ``macos-latest`` images work fine).
* The issue started happening two days ago with [this job](https://github.com/stephane-caron/qpsolvers/actions/runs/3767024165), which is a minor change following a successful build.
* Previously [this job](https://github.com/stephane-caron/qpsolvers/actions/runs/3764423513/jobs/6398842899) worked fine. It installed ``tox-4.0.16`` and ``tox_gh_actions-2.12.0``.
* Version-pinning tox-gh-actions to 2.12.0 removes the symptom, as done e.g. in [this job](https://github.com/stephane-caron/qpsolvers/actions/runs/3781004586). It installs ``tox-3.28.0`` and ``tox_gh_actions-2.12.0``.

**Comment by stephane-caron** [comment, marked as answer]
Actually the cure seems to come from tox being downgraded from >=4 to 3.28.0. [This job](https://github.com/stephane-caron/qpsolvers/actions/runs/3781594265/jobs/6428599575) was successful by version-pinning tox to 3.28.0, with no version pin on tox-gh-actions (that is, it installed ``tox==3.28.0`` and ``tox_gh_actions-3.0.0``).

Closing this issue in favor of https://github.com/tox-dev/tox/issues/2778.

**Comment by ymyzk** [comment]
@stephane-caron, FYI, we now need to install a different version of tox-gh-actions depending on the tox version.

- tox-gh-actions v2 must be used with tox v3 
- tox-gh-actions v3 must be used with tox v4

Please see https://github.com/ymyzk/tox-gh-actions#versions-and-compatibility for more details.
